### ISSUE TITLE
BUG?: maybe-incorrect reading large datetimes from SAS

### ISSUE THREAD

**Issue by jbrockmendel** [WizardMac/ReadStat#306]
TLDR: running down a maybe-bug in pandas where I have confirmed the pandas behavior matches the readstat behavior, but have not confirmed the _correct_ behavior.

I am trying to run down a 2020-era [comment](https://github.com/pandas-dev/pandas/pull/28047/files#diff-fd2c333eac306663e9e2de5be6e914aa8f35b1135ccc2301a19b7f76359f7802R232) about a bug in the pandas SAS reader:

```
    # NB. max datetime in SAS dataset is 31DEC9999:23:59:59.999
    #    but this is read as 29DEC9999:23:59:59.998993 by a buggy
    #    sas7bdat module
```

I have confirmed that reading the [file](https://github.com/pandas-dev/pandas/pull/28047/files#diff-6049785f4d66e175f3357b0f865a9dc3eda133e89a9b66bc071762cd7f13d3df) with pandas gets the Dec 29 datetime instead of the Dec 31 datetime.  I have also confirmed that using pyreadstat (which wraps readstat) returns a Dec 29 datetime (though it rounds down the seconds).

What I haven't been able to do is read the file with SAS to confirm that the comment is correct about the expected date being Dec 31 (or whether the fractional second should be truncated).

**Comment by evanmiller** [comment]
There's not much I can do about a maybe-bug, so I'm going to close this issue. Feel free to re-open if you can confirm that a discrepancy exists between ReadStat and SAS.

**Comment by jbrockmendel** [comment]
https://github.com/pandas-dev/pandas/issues/56014 got confirmation that SAS produces a different results from both pandas and readstat.

**Comment by jbrockmendel** [comment]
Figured out that SAS uses a calendar that treats 4000 and 8000 as not-leapyears.  See https://github.com/pandas-dev/pandas/pull/64344
